feat(version-source): mirror setuptools_scm override env vars for version

millsks opened the pull request:
## Description
We use hatch-vcs in all of our projects. There is a specific version format that they push to us on our build pipeline to align with their SCM reporting. The HATCH_VERSION_OVERRIDE variable will allow me to set that prior to our build so that their requirement is being used.

If this variable is not set then then default behavior takes over.

millsks committed:
feat(version-source): support HATCH_VERSION_OVERRIDE env var

millsks committed:
chore(release): bump version to 0.5.1

millsks committed:
feat(version-source): enhance get_version_data with environment variable support and add tests

millsks commented:
@ofek ready for review.

RonnyPfannschmidt reviewed (commented):
I think we should mirror the env vars setuptools_scm has for overrides/pretend versions

millsks committed:
feat(version-source): mirror setuptools_scm override env vars for version

Support SETUPTOOLS_SCM_PRETEND_VERSION and SETUPTOOLS_SCM_PRETEND_VERSION_FOR_<DIST_NAME> for version override, aligning with setuptools_scm conventions as requested in PR review. Also update tests to verify override precedence and compatibility.

millsks committed:
feat(version-source): use HATCH_VCS env vars for version override

Switch override environment variables from SETUPTOOLS_SCM to HATCH_VCS for version control, as requested in PR review. Tests updated to match new variable names and precedence.

millsks commented:
Thank you for the feedback @RonnyPfannschmidt. Do the changes that I made align with your suggestion? Let me know if it should be altered even further.
